identity-apps-core/apps/recovery-portal/src/main/webapp/password-recovery-confirm.jsp

```diff
@@ -54,15 +55,28 @@
 
     String confirmationKey = request.getParameter("confirmation");
     String callback = request.getParameter("callback");
+    String sp = request.getParameter("sp");
```

Comment:
et's encode these parameter values using OWASP Encoder; otherwise, these parameter usages can be exploited as XSS vulnerabilities.